app/controllers/changesets_controller.rb

```diff
@@ -113,21 +116,28 @@ def show
   #------------------------------------------------------------
 
   ##
-  # if a bounding box was specified do some sanity checks.
   # restrict changesets to those enclosed by a bounding box
-  def conditions_bbox(changesets, bbox)
-    if bbox
-      bbox.check_boundaries
-      bbox = bbox.to_scaled
-
-      changesets.where("min_lon < ? and max_lon > ? and min_lat < ? and max_lat > ?",
-                       bbox.max_lon.to_i, bbox.min_lon.to_i,
-                       bbox.max_lat.to_i, bbox.min_lat.to_i)
-    else
+  def conditions_bbox(changesets, min_lon, min_lat, max_lon, max_lat)
+    db_min_lat = (min_lat * GeoRecord::SCALE).to_i
+    db_max_lat = (max_lat * GeoRecord::SCALE).to_i
+    db_min_lon = (wrap_lon(min_lon) * GeoRecord::SCALE).to_i
+    db_max_lon = (wrap_lon(max_lon) * GeoRecord::SCALE).to_i
+
+    changesets = changesets.where("min_lat < ? and max_lat > ?", db_max_lat, db_min_lat)
+
+    if max_lon - min_lon >= 360
       changesets
+    elsif db_min_lon > db_max_lon
+      changesets.where("min_lon < ? or max_lon > ?", db_max_lon, db_min_lon)
+    else
+      changesets.where("min_lon < ? and max_lon > ?", db_max_lon, db_min_lon)
     end
   end
 
+  def wrap_lon(lon)
+    ((lon + 180) % 360) - 180
```

Comment:
I don't think this works? If we consider a range from -270 to 270 and apply this to both then we wind up with 90 to -90 which is clearly nonsense?

Even considering a simple globe spanning -180 to 180 wraps both to -180 leaving us with an empty box...

Reply:
> I don't think this works?

There are branches for "nonsense" cases.

> If we consider a range from -270 to 270 and apply this to both then we wind up with 90 to -90 which is clearly nonsense?

```
elsif db_min_lon > db_max_lon
```

> Even considering a simple globe spanning -180 to 180 wraps both to -180 leaving us with an empty box...

```
if max_lon - min_lon >= 360
```

Reply:
Added tests for -270 to 270 and -180 to 180.

Also added `:bbox` argument to changesets factory to shorten `create(:changeset, ...)` in tests.